muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java

```diff
@@ -2180,14 +2180,19 @@ else if (workspaceStudentId != null) {
           userEntity = userEntityFromWorkspaceUser;
         }
       }
-      entries = workspaceJournalController.listEntriesByWorkspaceEntityAndUserEntity(workspaceEntity, userEntity);
+      entries = workspaceJournalController.listEntriesByWorkspaceEntityAndUserEntity(workspaceEntity, userEntity, firstResult, maxResults);
     }
     
     for (WorkspaceJournalEntry entry : entries) {
+      UserEntity entryUserEntity = userEntityController.findUserEntityById(entry.getUserEntityId());
+      User user = userController.findUserByUserEntityDefaults(entryUserEntity);
```

Comment:
Will most likely cause a call to Pyramus but I assume moderate page size means this won't cause performance issues?